src/components/MapView/MapView.tsx

```diff
@@ -29,29 +29,35 @@ const MapView = forwardRef<MapViewHandle, ComponentProps>(
         const [isIdle, setIsIdle] = useState(false);
         const [currentPosition, setCurrentPosition] = useState(cachedUserLocation);
         const [userInteractedWithMap, setUserInteractedWithMap] = useState(false);
+        const hasAskedForLocationPermission = useRef(false);
+
+        const getUserLocationFromCache = useCallback(() => {
+            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+            if (cachedUserLocation || !initialState) {
+                return;
+            }
+
+            setCurrentPosition({longitude: initialState.location[0], latitude: initialState.location[1]});
+        }, [cachedUserLocation, initialState]);
 
         useFocusEffect(
             useCallback(() => {
                 if (isOffline) {
                     return;
                 }
 
-                getCurrentPosition(
-                    (params) => {
-                        const currentCoords = {longitude: params.coords.longitude, latitude: params.coords.latitude};
-                        setCurrentPosition(currentCoords);
-                        setUserLocation(currentCoords);
-                    },
-                    () => {
-                        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-                        if (cachedUserLocation || !initialState) {
-                            return;
-                        }
-
-                        setCurrentPosition({longitude: initialState.location[0], latitude: initialState.location[1]});
-                    },
-                );
-            }, [cachedUserLocation, initialState, isOffline]),
+                if (hasAskedForLocationPermission.current) {
+                    getUserLocationFromCache();
+                    return;
+                }
```

Comment:
Why would we need to call `getUserLocationFromCache` again? We should set the initial state only at first render

Reply:
I thought there was a use case for it but now I tested and I don't find any. Deleting this.